esp-hal-embassy/src/lib.rs

```diff
@@ -75,20 +75,10 @@ trait IntoAnyTimer: Into<AnyTimer> {}
 
 impl IntoAnyTimer for AnyTimer {}
 
-impl<T, DM> IntoAnyTimer for TimgTimer<T, DM>
-where
-    DM: esp_hal::Mode,
-    Self: Into<AnyTimer>,
-{
-}
+impl<T: esp_hal::timer::timg::Instance> IntoAnyTimer for TimgTimer<T> where Self: Into<AnyTimer> {}
```

Comment:
In the other peripherals, Instance implies `Into<AnyInstance>` (+ `'static`). I think we can do the same to timers as well. That may also mean `IntoAnyTimer` is meaningless, though I don't quite remember why I needed it in the first place - probably blanket impl'ing TimerCollection runs into some coherence rules.

Reply:
Hmm we can't add `+ Into<AnyInstance> + 'static` because it breaks peripheral ref on the low level timers, i.e `Alarm::new(&mut systimer.comparator0)`. I suppose these being "low level" timers they probably shouldn't being doing any peripheral ref stuff... but It's not a battle I have the energy for right now. I'll file an issue to resolve this later.